design/appendix/attack-scenarios.tex

```diff
@@ -0,0 +1,163 @@
+\chapter{Attack scenarios}\label{chap:attack scenarios}
+
+The goal of this \namecref{chap:attack scenarios} is to give an overview how various aspects of the Peras rules/requirements are necessicated by concrete attacks that could occur otherwise.
+\todo{Figure out if/in what capacity this should end up in the final version}
+
+\section{Simple attacks}
+
+\subsection{Abstaining during voting}\label{sec:abstaining during voting}
+
+An adversary with $\alpha$ stake will have $\alpha\cdot\perasN$ votes on average, so if they simply abstain, the honest nodes will likely fail to reach quorum even when they all vote for the same block when \[ \alpha \ge 1-\frac\perasQuorum\perasN = 0.25 \; \]
+Therefore, Peras doesn't provide fast settlement in the presence of an adversary with $\alpha\ge 25\%$.
+
+As usual, this is assuming full honest participation; smaller participation levels can only tolerate proportionally weaker adversaries.
+
+\subsection{Honest vote splitting}\label{sec:honest vote splitting}
+
+\todo{Mention Russell's suggested modification here}
+In the pre-alpha version of Peras, honest nodes vote for the newest block on their selection that is at least \perasBlockMinSlots{} slots old.
+It is relatively easy for an adversary to cause different honest nodes to vote for different blocks.
+
+For example, this allows an adversary with $\alpha<0.25$ to prevent quorum formation with decent probability (which they couldn't do by simply abstaining) by splitting the honest votes such that neither block gets \perasQuorum{} votes, causing the system to enter a cooldown period.
+Future versions of Peras will improve on this front by adding a dedicated \emph{pre-agreement} mechanism that \emph{avoids} a cooldown in this scenario.
+
+Write $\operatorname{tip}(C)$ for the tip of a chain $C$, and $\operatorname{trunc}(s,C)$ for the prefix of $C$ up until (but excluding) slot $s$.
+
+Concretely, consider the voting phase at the beginning of round $r$ starting in slot $s$, with all honest nodes having selected chain $C$ just before.
+
+Suppose that the adversary with stake $\alpha$ has a chain $D$ that is preferrable to $C$, and $V_C \neq V_D$ where \[V_c = \operatorname{tip}(\operatorname{trunc}(s-\perasBlockMinSlots,c))\] for $c\in\{C,D\}$.
+If the adversary now diffuses $D$ to only a subset of the honest nodes right before $r$ starts, these nodes will vote for $V_D$, while the rest will vote for $V_D$.
+\todo{Add some TikZ drawings}
+
+We can calculate a lower bound on the probability that an adversary has such a chain $D$ by considering a particular outcome of the leader schedule that guarantees the existence of $D$ when the adversary simply abstains while honest chain is growing before round $r$.
+
+\begin{enumerate}
+\item\label{enumi:honest vote splitting:a}
+  There is at least one adversarial active slot in the interval $(h,s-\perasBlockMinSlots)$ where $h$ is the last honest active slot before $s-\perasBlockMinSlots$.
```

Comment:
Hum, that's not the only thing, is it? The adversary needs either a block after the last honest block (which you describe with this sentence), or already a competing chain that is better, typically just one block that wins over the honest one in a tiebreaker. In this last case, the adversarial active slot could be older than the last honest active slot.

Isn't the case that the adversary has a competing chain that forks of the honest chain before `s - \perasBlockMinSlots` and contains at least one block before `s - \perasBlockMinSlots`?

Reply:
Ah:

> We stress that this is not the only scenario where the attack can be executed; for example, we have not considered tiebreakers and honest short forks.

OK nevermind!